src/main/java/com/minecolonies/core/colony/buildings/AbstractBuilding.java

```diff
@@ -1897,10 +1895,21 @@ public final ImmutableCollection<IRequestResolver<?>> getResolvers()
         final IStandardRequestManager requestManager = (IStandardRequestManager) colony.getRequestManager();
         if (!requestManager.getProviderHandler().getRegisteredResolvers(this).isEmpty())
         {
-            return ImmutableList.copyOf(requestManager.getProviderHandler().getRegisteredResolvers(this)
-              .stream()
-              .map(token -> requestManager.getResolverHandler().getResolver(token))
-              .collect(Collectors.toList()));
+            List<IRequestResolver<? extends IRequestable>> list = new ArrayList<>();
+            for (Iterator<IToken<?>> iterator = requestManager.getProviderHandler().getRegisteredResolvers(this).iterator(); iterator.hasNext(); )
+            {
+                final IToken<?> token = iterator.next();
+                try
+                {
+                    IRequestResolver<? extends IRequestable> resolver = requestManager.getResolverHandler().getResolver(token);
+                    list.add(resolver);
+                }
+                catch (Exception e)
```

Comment:
No idea if its safe to remove from registered resolvers, but shouldnt we log the exception?

Reply:
we could log it, but its just a "resolver not found for token" one, doesnt hold much info on why it happens